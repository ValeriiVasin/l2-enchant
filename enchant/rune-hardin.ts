import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const runeHardinEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Drop,
  enchantItem: Item.RuneHardin_1,
  chances: [60, 55, 50, 45, 45, 40, 35, 30, 25, 35, 30, 25, 20, 15, 25, 20, 15, 10, 5],
  items: [
    Item.RuneHardin_1,
    Item.RuneHardin_2,
    Item.RuneHardin_3,
    Item.RuneHardin_4,
    Item.RuneHardin_5,
    Item.RuneHardin_6,
    Item.RuneHardin_7,
    Item.RuneHardin_8,
    Item.RuneHardin_9,
    Item.RuneHardin_10,
    Item.RuneHardin_11,
    Item.RuneHardin_12,
    Item.RuneHardin_13,
    Item.RuneHardin_14,
    Item.RuneHardin_15,
    Item.RuneHardin_16,
    Item.RuneHardin_17,
    Item.RuneHardin_18,
    Item.RuneHardin_19,
    Item.RuneHardin_20,
  ],
  boundaries: new Set([Item.RuneHardin_6, Item.RuneHardin_11, Item.RuneHardin_16]),
});
