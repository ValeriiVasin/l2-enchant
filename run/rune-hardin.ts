import { run } from '../helpers/run';
import { Item } from '../items';

run(
  Item.RuneHardin_1,
  [
    Item.RuneHardin_2,
    Item.RuneHardin_3,
    Item.RuneHardin_4,
    Item.RuneHardin_5,
    Item.RuneHardin_6,
    Item.RuneHardin_7,
    Item.RuneHardin_8,
    Item.RuneHardin_9,
    Item.RuneHardin_10,
    Item.RuneHardin_11,
    Item.RuneHardin_12,
    Item.RuneHardin_13,
    Item.RuneHardin_14,
    Item.RuneHardin_15,
    Item.RuneHardin_16,
    Item.RuneHardin_17,
    Item.RuneHardin_18,
    Item.RuneHardin_19,
    Item.RuneHardin_20,
  ],
  100_000,
);
