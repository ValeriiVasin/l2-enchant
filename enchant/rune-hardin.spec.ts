import { runeHardinEnchantMap } from './rune-hardin';

it('enchant map: rune hardin', () => {
  expect(runeHardinEnchantMap).toMatchSnapshot();
});
