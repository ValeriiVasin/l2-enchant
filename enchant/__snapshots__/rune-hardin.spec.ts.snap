// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: rune hardin 1`] = `
Map {
  "Кристалл Души Хардина Ур.1" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.1",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Кристалл Души Хардина Ур.2",
      },
      Object {
        "chance": 40,
        "item": "Кристалл Души Хардина Ур.1",
      },
    ],
  },
  "Кристалл Души Хардина Ур.2" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.2",
    "results": Array [
      Object {
        "chance": 55,
        "item": "Кристалл Души Хардина Ур.3",
      },
      Object {
        "chance": 45,
        "item": "Кристалл Души Хардина Ур.1",
      },
    ],
  },
  "Кристалл Души Хардина Ур.3" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.3",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Кристалл Души Хардина Ур.4",
      },
      Object {
        "chance": 50,
        "item": "Кристалл Души Хардина Ур.1",
      },
    ],
  },
  "Кристалл Души Хардина Ур.4" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.4",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Кристалл Души Хардина Ур.5",
      },
      Object {
        "chance": 55,
        "item": "Кристалл Души Хардина Ур.1",
      },
    ],
  },
  "Кристалл Души Хардина Ур.5" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.5",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Кристалл Души Хардина Ур.6",
      },
      Object {
        "chance": 55,
        "item": "Кристалл Души Хардина Ур.1",
      },
    ],
  },
  "Кристалл Души Хардина Ур.6" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.6",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Кристалл Души Хардина Ур.7",
      },
      Object {
        "chance": 60,
        "item": "Кристалл Души Хардина Ур.6",
      },
    ],
  },
  "Кристалл Души Хардина Ур.7" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.7",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Кристалл Души Хардина Ур.8",
      },
      Object {
        "chance": 65,
        "item": "Кристалл Души Хардина Ур.6",
      },
    ],
  },
  "Кристалл Души Хардина Ур.8" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.8",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Кристалл Души Хардина Ур.9",
      },
      Object {
        "chance": 70,
        "item": "Кристалл Души Хардина Ур.6",
      },
    ],
  },
  "Кристалл Души Хардина Ур.9" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.9",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Кристалл Души Хардина Ур.10",
      },
      Object {
        "chance": 75,
        "item": "Кристалл Души Хардина Ур.6",
      },
    ],
  },
  "Кристалл Души Хардина Ур.10" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.10",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Кристалл Души Хардина Ур.11",
      },
      Object {
        "chance": 65,
        "item": "Кристалл Души Хардина Ур.6",
      },
    ],
  },
  "Кристалл Души Хардина Ур.11" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.11",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Кристалл Души Хардина Ур.12",
      },
      Object {
        "chance": 70,
        "item": "Кристалл Души Хардина Ур.11",
      },
    ],
  },
  "Кристалл Души Хардина Ур.12" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.12",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Кристалл Души Хардина Ур.13",
      },
      Object {
        "chance": 75,
        "item": "Кристалл Души Хардина Ур.11",
      },
    ],
  },
  "Кристалл Души Хардина Ур.13" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.13",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Кристалл Души Хардина Ур.14",
      },
      Object {
        "chance": 80,
        "item": "Кристалл Души Хардина Ур.11",
      },
    ],
  },
  "Кристалл Души Хардина Ур.14" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.14",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Кристалл Души Хардина Ур.15",
      },
      Object {
        "chance": 85,
        "item": "Кристалл Души Хардина Ур.11",
      },
    ],
  },
  "Кристалл Души Хардина Ур.15" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.15",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Кристалл Души Хардина Ур.16",
      },
      Object {
        "chance": 75,
        "item": "Кристалл Души Хардина Ур.11",
      },
    ],
  },
  "Кристалл Души Хардина Ур.16" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.16",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Кристалл Души Хардина Ур.17",
      },
      Object {
        "chance": 80,
        "item": "Кристалл Души Хардина Ур.16",
      },
    ],
  },
  "Кристалл Души Хардина Ур.17" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.17",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Кристалл Души Хардина Ур.18",
      },
      Object {
        "chance": 85,
        "item": "Кристалл Души Хардина Ур.16",
      },
    ],
  },
  "Кристалл Души Хардина Ур.18" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.18",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Кристалл Души Хардина Ур.19",
      },
      Object {
        "chance": 90,
        "item": "Кристалл Души Хардина Ур.16",
      },
    ],
  },
  "Кристалл Души Хардина Ур.19" => Object {
    "cost": "Кристалл Души Хардина Ур.1",
    "item": "Кристалл Души Хардина Ур.19",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Кристалл Души Хардина Ур.20",
      },
      Object {
        "chance": 95,
        "item": "Кристалл Души Хардина Ур.16",
      },
    ],
  },
}
`;
